src/DotNetty.Codecs.Http/HttpServerUpgradeHandler.cs

```diff
@@ -254,34 +254,28 @@ bool Upgrade(IChannelHandlerContext ctx, IFullHttpRequest request)
             var upgradeEvent = new UpgradeEvent(upgradeProtocol, request);
 
             IUpgradeCodec finalUpgradeCodec = upgradeCodec;
-            ctx.WriteAndFlushAsync(upgradeResponse).ContinueWith(t =>
-                {
-                    try
-                    {
-                        if (t.Status == TaskStatus.RanToCompletion)
-                        {
-                            // Perform the upgrade to the new protocol.
-                            this.sourceCodec.UpgradeFrom(ctx);
-                            finalUpgradeCodec.UpgradeTo(ctx, request);
-
-                            // Notify that the upgrade has occurred. Retain the event to offset
-                            // the release() in the finally block.
-                            ctx.FireUserEventTriggered(upgradeEvent.Retain());
-
-                            // Remove this handler from the pipeline.
-                            ctx.Channel.Pipeline.Remove(this);
-                        }
-                        else
-                        {
-                            ctx.Channel.CloseAsync();
-                        }
-                    }
-                    finally
-                    {
-                        // Release the event if the upgrade event wasn't fired.
-                        upgradeEvent.Release();
-                    }
-                }, TaskContinuationOptions.ExecuteSynchronously);
+            try
+            {
+                Task writeTask = ctx.WriteAndFlushAsync(upgradeResponse);
+                
+                // Perform the upgrade to the new protocol.
+                this.sourceCodec.UpgradeFrom(ctx);
```

Comment:
@StormHub, check this out. I change it to align with netty. So upgrade doesn't wait write to complete. 
https://github.com/netty/netty/blob/00afb19d7a37de21b35ce4f6cb3fa7f74809f2ab/codec-http/src/main/java/io/netty/handler/codec/http/HttpServerUpgradeHandler.java#L324